src/launch/service.c

```diff
@@ -423,6 +437,33 @@ static int service_watch_unit(Service *service, const char *unit) {
                 NULL,
                 service
         );
+        if (error_trace(r) < 0)
+                goto error;
+
+        return 1;
+
+error:
+        service_reset_activation(service);
+        return 1;
+}
+
+static int service_watch_unit(Service *service, const char *unit) {
+        int r;
+
+        assert(!service->slot_watch_unit);
+
+        r = sd_bus_call_method_async(
+                service->launcher->bus_regular,
+                &service->slot_watch_unit,
+                "org.freedesktop.systemd1",
+                "/org/freedesktop/systemd1",
+                "org.freedesktop.systemd1.Manager",
+                "LoadUnit",
+                service_watch_unit_callback,
+                service,
+                "s",
+                unit
+        );
```

Comment:
What happens if systemd reloads configuration between our call to `LoadUnit` and the following watch-operation? That is, if the unit in question changes exactly at that moment, we will possible watch the wrong path, right? Not a big issue, as this is inherent to the way systemd-reloads work, but I was just wondering what the story there is.

Oh, and same issue arises if the unit is modified at any time while we watch it. We will never re-resolve the object-path, so any changes to aliases will be ignored by us. Not really related to your patch, as the old code already suffered from those problems, but definitely something to keep in mind.